src/main/java/io/vertx/core/http/impl/ClientHandler.java

```diff
@@ -111,8 +110,11 @@ protected void handleMessage(ClientConnection conn, ContextImpl context, Channel
           break;
         case PING:
           // Echo back the content of the PING frame as PONG frame as specified in RFC 6455 Section 5.5.2
-          chctx.writeAndFlush(new WebSocketFrameImpl(FrameType.PONG, frame.getBinaryData()));
```

Comment:
I think this failed to work, because of encoding.

ChannelHandlerContext doesn't have encoding for the vertx Frame implementations.

The fix takes advantage of the encoder on the connection:
https://github.com/eclipse/vert.x/blob/master/src/main/java/io/vertx/core/http/impl/Http1xConnectionBase.java#L52

Reply:
I think the same problem may exist a few lines down under `case CLOSE:`